proposals/0271-swiftpm-binary-dependencies.md

```diff
@@ -0,0 +1,417 @@
+  imposes certain requirements on the design which make it more complex than the
+  existing proposal. In particular, it means that the metadata about how the
+  source and artifacts are mapped must be kept somewhere adjacent to but
+  distinct from the package description (since a source package needs to define
+  its source layout). However, such a mechanism must also be defined in a way
+  that works when no source layout is present to support binary only packages.
+  
+  Finally, since it would be a feature with user-authored metadata, such a
+  mechanism would need to be updated when any other SwiftPM enhancement
+  introduces or changes the nature of the source layout specification.
+
+Taken together, the above points led us to focus on a proposal focused at
+"vendored binaries", while our hope is that artifact caching eventually becomes
+a built-in and automatic feature of the package manager which applies to all
+packages.
+
+### Binary Signatures
+
+We considered adding signature checks during the checkout of binary dependencies but when these have transitive dependencies it gets complicated expressing that in the `Package.swift`.
+
+```
+     let package = Package(
+         name: "Paper",
+         products: [...],
+         dependencies: [
+             .package(url: "http://some/other/lib", .exact("1.2.3"), binarySignature: .any),
+             .package(url: "http://some/other/lib", .exact("1.2.3"), binarySignature: .gpg("XXX")"),
+         ],
+         targets: [...]
+     )
+```
+
+### Binary target vs. binary product
+During the discussion, it was brought up wether a binary dependency should be declared as a target or a product. Below is a list of what we took into consideration when deciding between target and product. In the end, a target seems like a better choice for a binary dependency.
+
+- Targets allow configuring linker/compiler flags; this ability might be necessary for static libraries
+- There are already `systemLibraryTargets` which are essentially dependencies on pre-existing binaries on the host system
+- Targets represent a single module, the same is true for one XCFramework
+- Currently there is no way to depend on products, so mixed binary and source packages might be harder if we go with a binary product approach
+- Analogy with what currently is being produced by source packages (dylibs => product)
+
+### .o file format
+During the discussion of the proposal, the idea of using `.o` files was brought up. This would follow what SwiftPM creates for source-based dependencies right now; therefor, making the integration potentially easier. The main benefit of using `.o` files would be that the product linking the binary artifact can decide wether to link it dynamically or statically. However, further discussion needs to happen here how this would work. For now XCFrameworks are the initial format. XCFrameworks will allow current framework authors to package their existing XCFrameworks and distribute them via SwiftPM.
```

Comment:
```suggestion
During the discussion of the proposal, the idea of using `.o` files was brought up. This would follow what SwiftPM creates for source-based dependencies right now; therefore, making the integration potentially easier. The main benefit of using `.o` files would be that the product linking the binary artifact can decide whether to link it dynamically or statically. However, further discussion needs to happen here how this would work. For now XCFrameworks are the initial format. XCFrameworks will allow current framework authors to package their existing XCFrameworks and distribute them via SwiftPM.
```

(`therefor` => `therefore`; `wether` => `whether`)